src/software/thunderscope/robot_diagnostics/diagnostics_input_widget.py

```diff
@@ -23,53 +31,44 @@ class FullSystemConnectWidget(QWidget):
     """
 
     # Signal to indicate if manual controls should be disabled based on boolean parameter
-    toggle_controls_signal = pyqtSignal(bool)
 
-    def __init__(self) -> None:
+    def __init__(
+        self, on_control_mode_switch_callback: Callable[[ControlMode], None]
+    ) -> None:
         """
         Initialises a new Fullsystem Connect Widget to allow switching between Diagnostics and XBox control
+        :param on_control_mode_switch_callback The signal to use for handling changes in input mode
         """
-
-        super(FullSystemConnectWidget, self).__init__()
+        super(DiagnosticsInputToggleWidget, self).__init__()
 
         vbox_layout = QVBoxLayout()
-        self.connect_options_group = QButtonGroup()
 
-        radio_button_names = ["Diagnostics Control", "XBox Control"]
+        self.connect_options_group = QButtonGroup()
 
         self.connect_options_box, self.connect_options = common_widgets.create_radio(
-            radio_button_names, self.connect_options_group
+            ["Diagnostics Control", "Handheld Control"], self.connect_options_group
         )
 
         self.diagnostics_control_button = self.connect_options[ControlMode.DIAGNOSTICS]
-        self.xbox_control_button = self.connect_options[ControlMode.XBOX]
+        self.handheld_control_button = self.connect_options[ControlMode.HANDHELD]
 
         self.diagnostics_control_button.clicked.connect(
-            lambda: self.switch_control_mode(ControlMode.DIAGNOSTICS)
+            lambda: on_control_mode_switch_callback(ControlMode.DIAGNOSTICS)
         )
-        self.xbox_control_button.clicked.connect(
-            lambda: self.switch_control_mode(ControlMode.XBOX)
+        self.handheld_control_button.clicked.connect(
+            lambda: on_control_mode_switch_callback(ControlMode.HANDHELD)
         )
 
+        self.handheld_control_button.setEnabled(False)
         self.diagnostics_control_button.setChecked(True)
-        self.control_mode = ControlMode.DIAGNOSTICS
 
         vbox_layout.addWidget(self.connect_options_box)
 
         self.setLayout(vbox_layout)
 
-    def switch_control_mode(self, mode: ControlMode) -> None:
-        """
-        Switches the control mode to the given mode
-
-        Emits a signal to indicate whether diagnostics controls should be disabled or not
-
-        :param mode: mode to switch to (one of ControlMode values)
-
-        """
-        self.control_mode = mode
-
-        self.toggle_controls_signal.emit(self.control_mode == ControlMode.DIAGNOSTICS)
-
-    def refresh(self) -> None:
-        pass
+    def refresh(self, status: ControllerConnectionState) -> None:
+        if status == ControllerConnectionState.DISCONNECTED:
+            self.diagnostics_control_button.click()
+        self.handheld_control_button.setEnabled(
+            status == ControllerConnectionState.CONNECTED
+        )
```

Comment:
this logic looks a little funky, can you explain it?

also missing docs

Reply:
It's got to do with handling a controller being unplugged, and automatically switching back to UI diagnostics control (looks like I needed the callback to run). There's probably a way to make it less funky, will look into it.